File: ITMO.Python/PythonProject/Demo02/benchmark/csharp/csharp/Starter.cs
using System.Numerics;
using csharp.Dto;
using csharp.Engine;

namespace csharp
{
    internal class Starter
    {
        private const string TYPE_INT = "Integer";
        private const string TYPE_DOUBLE = "Double";
        private const string TYPE_BIG_INTEGER = "BigInteger";

        private string _inputPath;
        private string _dataType;

        public Starter(string inputPath, string dataType)
        {
            _inputPath= inputPath;
            _dataType = dataType;
        }

        public async Task<double> StartBenchmark()
        {
            switch (_dataType)
            {
                case TYPE_INT:
                    return await StartIntBench();
                case TYPE_DOUBLE:
                    return await StartDoubleBench();
                case TYPE_BIG_INTEGER:
                    return await StartBigIntBench();
                default:
                    Console.WriteLine("Неверный тип данных");
                    return -999;
            }
        }

        private async Task<double> StartBigIntBench()
        {
            JSONOperator<BigInteger> json = new JSONOperator<BigInteger>(_inputPath);
            InputData<BigInteger> inputData = await json.ReadJsonAsync();

            if (inputData is null)
            {
                Console.WriteLine($"[C#] Файл по пути {_inputPath} не найден");
                return -999;
            }

            int startCount = inputData.StartCount;

            Benchmark<BigInteger, BigIntMathematics> benchmark = new Benchmark<BigInteger, BigIntMathematics>(
                        inputData.MatrixA, inputData.MatrixB,
                        inputData.TransA, inputData.TransB,
                        inputData.Alfa, inputData.Betta, inputData.ThreadCount);

            for (int idx = 0; idx < startCount; idx++)
            {
                var time = new System.Diagnostics.Stopwatch();
                time.Start();
                benchmark.Dgemm();
                time.Stop();
                benchmark.AddTimeInterval((double)time.ElapsedMilliseconds);
            }
            return benchmark.TimeInterval.Sum();
        }

        private async Task<double> StartIntBench()
        {
            JSONOperator<int> json = new JSONOperator<int>(_inputPath);
            InputData<int> inputData = await json.ReadJsonAsync();

            if (inputData is null)
            {
                Console.WriteLine($"[C#] Файл по пути {_inputPath} не найден");
                return -999;
            }

            int startCount = inputData.StartCount;

            Benchmark<int, IntMathematics> benchmark = new Benchmark<int, IntMathematics>(
                        inputData.MatrixA, inputData.MatrixB,
                        inputData.TransA, inputData.TransB,
                        inputData.Alfa, inputData.Betta, inputData.ThreadCount);

            for (int idx = 0; idx < startCount; idx++)
            {
                var time = new System.Diagnostics.Stopwatch();
                time.Start();
                benchmark.Dgemm();
                time.Stop();
                benchmark.AddTimeInterval((double)time.ElapsedMilliseconds);
            }
            return benchmark.TimeInterval.Sum();
        }

        private async Task<double> StartDoubleBench()
        {
            JSONOperator<double> json = new JSONOperator<double>(_inputPath);
            InputData<double> inputData = await json.ReadJsonAsync();

            if (inputData is null)
            {
                Console.WriteLine($"[C#] Файл по пути {_inputPath} не найден");
                return -999;
            }

            int startCount = (int)inputData.StartCount;

            Benchmark<double, DoubleMathematics> benchmark = new Benchmark<double, DoubleMathematics>(
                        inputData.MatrixA, inputData.MatrixB,
                        inputData.TransA, inputData.TransB,
                        inputData.Alfa, inputData.Betta, inputData.ThreadCount);

            for (int idx = 0; idx < startCount; idx++)
            {
                var time = new System.Diagnostics.Stopwatch();
                time.Start();
                benchmark.Dgemm();
                time.Stop();
                benchmark.AddTimeInterval((double)time.ElapsedMilliseconds);
            }
            return benchmark.TimeInterval.Sum();
        }
    }
}

File: ITMO.Python/PythonProject/Demo02/benchmark/java/src/Starter.java
import Dto.InputData;
import Engine.Benchmark;
import com.fasterxml.jackson.core.JsonProcessingException;

import java.math.BigInteger;
import java.util.Arrays;

public class Starter {
    private final String TYPE_INTEGER = "Integer";
    private final String TYPE_DOUBLE = "Double";
    private final String TYPE_BIG_INTEGER = "BigInteger";

    private String inputPath;
    private String dataType;

    public Starter(String inputPath, String dataType) {
        this.inputPath = inputPath;
        this.dataType = dataType;
    }

    public double startBenchmark() {
        try {
            switch (dataType) {
                case (TYPE_INTEGER):
                    return startIntBench();
                case (TYPE_DOUBLE):
                    return startDoubleBench();
                case (TYPE_BIG_INTEGER):
                    return startBigIntBench();
                default:
                    System.out.println("[Java] Неверный тип данных");
                    return -999;
            }
        } catch (IllegalStateException e) {
            e.printStackTrace();
            return -999;
        } catch (JsonProcessingException e) {
            e.printStackTrace();
            return -999;
        } catch (NullPointerException e) {
            e.printStackTrace();
            System.out.println(e.getMessage());
            return -999;
        }
    }

    private double startBigIntBench() throws IllegalStateException, JsonProcessingException, NullPointerException {
        JSONOperator<BigInteger> json = new JSONOperator(inputPath, BigInteger.class);
        InputData<BigInteger> inputData = json.readJSON();

        Benchmark<BigInteger> benchmark = new Benchmark(inputData, BigInteger.class);

        for (int idx = 0; idx < inputData.getStartCount(); idx++) {
            long startTime = System.nanoTime();
            benchmark.dgemm();
            benchmark.setValueTimeInterval(idx,
                    ((double) System.nanoTime() - (double) startTime) / Math.pow(10, 6));
        }

        return Arrays.stream(benchmark.getTimeInterval()).sum();
    }

    private double startIntBench() throws IllegalStateException, JsonProcessingException, NullPointerException {
        JSONOperator<Integer> json = new JSONOperator(inputPath, Integer.class);
        InputData<Integer> inputData = json.readJSON();

        Benchmark<Integer> benchmark = new Benchmark(inputData, Integer.class);

        for (int idx = 0; idx < inputData.getStartCount(); idx++) {
            long startTime = System.nanoTime();
            benchmark.dgemm();
            benchmark.setValueTimeInterval(idx,
                    ((double) System.nanoTime() - (double) startTime) / Math.pow(10, 6));
        }

        return Arrays.stream(benchmark.getTimeInterval()).sum();
    }

    private double startDoubleBench() throws IllegalStateException, JsonProcessingException, NullPointerException {
        JSONOperator<Double> json = new JSONOperator(inputPath, Double.class);
        InputData<Double> inputData = json.readJSON();

        Benchmark<Double> benchmark = new Benchmark(inputData, Double.class);

        for (int idx = 0; idx < inputData.getStartCount(); idx++) {
            long startTime = System.nanoTime();
            benchmark.dgemm();
            benchmark.setValueTimeInterval(idx,
                    ((double) System.nanoTime() - (double) startTime) / Math.pow(10, 6));
        }

        return Arrays.stream(benchmark.getTimeInterval()).sum();
    }
}
